client/src/contexts/DataContext.js

```diff
@@ -0,0 +1,150 @@
+            for(let i=0; i<priceLowToHigh.length; i++){
+              if(typeof priceLowToHigh[i].name.charAt(0) === 'string'){
+                priceLowToHigh[i].name = priceLowToHigh[i].name.charAt(0).toUpperCase() + priceLowToHigh[i].name.slice(1)
+              }
+              coinNames.push(priceLowToHigh[i].name)
+            }
+          }
+          else if(sortState === 'highLow24h'){
+            let dailyHighLow = coinData.slice(0,10)
+            dailyHighLow.sort((a,b) => b.percent_change_24h - a.percent_change_24h)
+            coinList = dailyHighLow
+            for(let i=0; i<dailyHighLow.length; i++){
+              if(typeof dailyHighLow[i].name.charAt(0) === 'string'){
+                dailyHighLow[i].name = dailyHighLow[i].name.charAt(0).toUpperCase() + dailyHighLow[i].name.slice(1)
+              }
+              coinNames.push(dailyHighLow[i].name)
+            }
+          }
+          else if(sortState === 'lowHigh24h'){
+            let dailyLowToHigh = coinData.slice(0,10)
+            dailyLowToHigh.sort((a,b) => a.percent_change_24h - b.percent_change_24h)
+            coinList = dailyLowToHigh
+            for(let i=0; i<dailyLowToHigh.length; i++){
+              if(typeof dailyLowToHigh[i].name.charAt(0) === 'string'){
+                dailyLowToHigh[i].name = dailyLowToHigh[i].name.charAt(0).toUpperCase() + dailyLowToHigh[i].name.slice(1)
+              }
+              coinNames.push(dailyLowToHigh[i].name)
+            }
+          }
+          else if(sortState === 'highLow7d'){
+            let weeklyHighLow = coinData.slice(0,10)
+            weeklyHighLow.sort((a,b) => b.percent_change_7d - a.percent_change_7d)
+            coinList = weeklyHighLow
+            for(let i=0; i<weeklyHighLow.length; i++){
+              if(typeof weeklyHighLow[i].name.charAt(0) === 'string'){
+                weeklyHighLow[i].name = weeklyHighLow[i].name.charAt(0).toUpperCase() + weeklyHighLow[i].name.slice(1)
+              }
+              coinNames.push(weeklyHighLow[i].name)
+            }
+          }
+          else if(sortState === 'lowHigh7d'){
+            let weeklyLowHigh = coinData.slice(0,10)
+            weeklyLowHigh.sort((a,b) => a.percent_change_7d - b.percent_change_7d)
+            coinList = weeklyLowHigh
+            for(let i=0; i<weeklyLowHigh.length; i++){
+              if(typeof weeklyLowHigh[i].name.charAt(0) === 'string'){
+                weeklyLowHigh[i].name = weeklyLowHigh[i].name.charAt(0).toUpperCase() + weeklyLowHigh[i].name.slice(1)
+              }
+              coinNames.push(weeklyLowHigh[i].name)
+            }
+          }
+          else{
+            coinList = coinData
+            for(let i=0; i<coinData.length; i++){
+              if(typeof coinData[i].name.charAt(0) === 'string'){
+                coinData[i].name = coinData[i].name.charAt(0).toUpperCase() + coinData[i].name.slice(1)
+              }
+              coinNames.push(coinData[i].name)
+            }
+          }
```

Comment:
This ~80 lines of sorting code can definitely have less duplication with the use of helper functions.